src/auth/index.js

```diff
@@ -0,0 +1,127 @@
+import Vue from "vue";
+import createAuth0Client from "@auth0/auth0-spa-js";
+
+/** Define a default action to perform after authentication */
+const DEFAULT_REDIRECT_CALLBACK = () =>
+  window.history.replaceState({}, document.title, window.location.pathname);
+
+let instance;
+
+/** Returns the current instance of the SDK */
+export const getInstance = () => instance;
+
+/** Creates an instance of the Auth0 SDK. If one has already been created, it returns that instance */
+export const useAuth0 = ({
+  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK,
+  redirectUri = window.location.origin,
+  ...options
+}) => {
+  if (instance) return instance;
+
+  // The 'instance' is simply a Vue object
+  instance = new Vue({
+    data() {
+      return {
+        loading: true,
+        isAuthenticated: false,
+        user: {},
```

Comment:
Is this connected to vuex store? If not how are you going to get this information in other views. I guess information like username or email may be useful to show to user after login for example.

Reply:
For now, I add only pure base vue. Also all user data is stored under global $auth.user

Reply:
Oh so it's fine then